src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -60,6 +63,21 @@ def getNewTimestamp(doc):
     return subDoc
 
 
+def customDID(pname):
+    """
+    Helper function to create custom Name from given pileup one
+    """
+    pat = re.compile(".*-V[0-9]$")
+    if re.match(pat, pname):
+        arr = pname.split("-V")
+        ver = int(arr[-1])
```

Comment:
I'm not sure what do you mean by "natural version value", either provide specific pattern or further clarification how to adjust, e.g. version should be any number, or it can be any character starting with `V`. So far I'm adjusting it to `".*-V.*[0-9]$"` pattern, which reads `-V<any character><number>`. If you disagree please provide the pattern you like to be implemented.

Reply:
ok, now you provided concrete pattern and I adjusted the code and unit test accordingly.